### PR TITLE
Add INSTALL file

### PR DESCRIPTION
**Pull request by otyshchenko1**

### CODE REVIEW

File: INSTALL [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+3. Checked with Python v 2.7.12, but other should also work
+
+About:
+======
+Gen3-test product is based on Renesas's BSP and Xen hypervisor. It contains a set of features required to have
+both Display and GPU support in several domains at the same time (device passthrough, PVDRM, VGPU, etc).
+
+There are three domains which are running on top of Xen:
+1. Generic machine independent control domain named "domain 0". This "initramfs" based domain is responsible
+for managing VM only (create/destroy/reboot guest domains). There is no HW assigned to this domain.
```
Comment (andr2000) [inline INSTALL]: VM**s**

File: INSTALL [hunk 2]
```diff
@@ -0,0 +1,213 @@
+Prerequisites:
+==============
+1. Python Git: sudo apt-get install python-git
+2. Google repo: https://source.android.com/source/downloading
+3. Checked with Python v 2.7.12, but other should also work
+
+About:
+======
+Gen3-test product is based on Renesas's BSP and Xen hypervisor. It contains a set of features required to have
```
Comment (andr2000) [inline INSTALL]: Renesas's -> Renesas'

File: INSTALL [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+both Display and GPU support in several domains at the same time (device passthrough, PVDRM, VGPU, etc).
+
+There are three domains which are running on top of Xen:
+1. Generic machine independent control domain named "domain 0". This "initramfs" based domain is responsible
```
Comment (andr2000) [inline INSTALL]: not sure why you need quotes around initramfs

**Comment by aanisov** [comment]
I would suggest you just squash commits into one. Just add a proper instruction.

File: INSTALL [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+for managing VM only (create/destroy/reboot guest domains). There is no HW assigned to this domain.
+2. Machine dependent driver domain named "domain D". Almost all HW are assigned to this domain.
```
Comment (andr2000) [inline INSTALL]: "domain D" -> DomD?

File: INSTALL (hunk 1 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL (hunk 3 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+As this domain is not 1:1 mapped, Xen controls IPMMU to do a proper address translations (IPA to PA) for DMA masters.
+It also contains different PV backends running inside it in order to provide guests domains (without HW assigned)
```
Comment (andr2000) [inline INSTALL]: different PV backends - > different para-virtualized (PV) backends

Comment (andr2000) [reply INSTALL]: guests domains -> guest domains

File: INSTALL [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+with different services such as Audio, Display, Net, Block, etc.
+3. Machine dependent simple guest domain named "domain U". It has different PV frontends running and
+doesn't have any HW assigned except GPU. The latter are shared between "domain D" and "domain U" in current setup.
```
Comment (andr2000) [inline INSTALL]: The letter are -> The letter is

File: INSTALL (hunk 4 above)
Comment (otyshchenko1) [reply INSTALL]: Agree, will change to how they look in xl list:
Domain 0 -> Domain-0
domain D -> DomD
domain U -> DomU

File: INSTALL (hunk 5 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL (hunk 6 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+Build:
+======
+Our new build system uses set of additional meta layers and tools.
```
[the remark this answers is not in the document]
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+cp xt-build-server.cfg xt-gen3-test.cfg
+
+3. Edit it to fit your environment: the variables we need to change
+are under [path] section. Please set:
+ - workspace_base_dir: change it to point to the place where the build will happen
+ - workspace_storage_base_dir: change it to where downloads and other files will be put
+
+For example,
+
+workspace_base_dir = /tmp/workspace_base
```
Comment (andr2000) [inline INSTALL]: never put /tmp in examples for build target: people will use it as is and will loose all after reboot...

File: INSTALL [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+workspace_storage_base_dir = /tmp/workspace_storage_base
+
+4. Run the build script for current stable release:
+
+python ./build_prod.py --build-type dailybuild --machine MACHINE_NAME --product gen3-test --with-local-conf --config xt-gen3-test.cfg
+
+Here and everywhere down the document the supported MACHINE_NAMEs are:
```
Comment (andr2000) [inline INSTALL]: Here and everywhere == Hereafter

File: INSTALL [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+- salvator-x-h3  (Salvator-X board with H3 SoC installed)
+- salvator-x-m3  (Salvator-X board with M3 SoC installed)
+- salvator-x-m3n (Salvator-XS board with M3N SoC installed)
+
+5. After that you will have all the build environment setup at workspace_base_dir
+directory.
+
+6. Download and copy EVA proprietary archives on your local filesystem in some directory:
```
Comment (andr2000) [inline INSTALL]: archives -> packages

Comment (andr2000) [reply INSTALL]: Download and copy EVA proprietary archives on your local filesystem in some directory ->
Download and copy EVA proprietary packages to your local filesystem at some directory

File: INSTALL [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+[TODO: Should be updated]
```
Comment (andr2000) [inline INSTALL]: I would remove TODO and put "For example,"

File: INSTALL [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+workspace_base/build/conf/local.conf
+
+and append the following:
+
+XT_RCAR_EVAPROPRIETARY_DIR = "/path/to/eva/archives"
+
+with the correct path to the archives set.
+
+And
+XT_ALLOW_SSTATE_CACHE_MIRROR_USE = "1"
```
Comment (andr2000) [inline INSTALL]: Why? Do you expect users to have mirrors available?

File: INSTALL [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+
+- For building Domain 0:
+cd workspace_base/build/tmp/work/x86_64-xt-linux/dom0-image-thin-initramfs/1.0-r0/repo/
+- For building Domain D:
+cd workspace_base/build/tmp/work/x86_64-xt-linux/domd-image-weston/1.0-r0/repo/
+- For building Domain U:
+cd workspace_base/build/tmp/work/x86_64-xt-linux/domu-image-weston/1.0-r0/repo/
+
+source poky/oe-init-build-env
+bitbake core-image-weston
```
Comment (andr2000) [inline INSTALL]: this differs for Dom0

Comment (andr2000) [reply INSTALL]: bitbake core-image-thin-initramfs

File: INSTALL (hunk 9 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL (hunk 10 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL (hunk 11 above)
Comment (otyshchenko1) [reply INSTALL]: ok

File: INSTALL (hunk 12 above)
Comment (otyshchenko1) [reply INSTALL]: I will drop

**Comment by andr2000** [comment]
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`

**Comment by otyshchenko1** [comment]
@iartemenko pls merge if you don't have objections regarding the document

**Comment by otyshchenko1** [comment]
@iartemenko ping

**Pull request merged**